src/VirtualClient/VirtualClient.Contracts.UnitTests/Parser/TextParserExtensionsTests.cs

```diff
@@ -34,17 +35,62 @@ public void TextParserExtensionsTranslateByteUnitAsExpected(string orginalText,
         [TestCase("100tb", MetricUnit.Gigabytes, "102400")]
         [TestCase("100pb", MetricUnit.Terabytes, "102400")]
         [TestCase("1tb", MetricUnit.Kilobytes, "1073741824")]
-        public void TextParserExtensionsTranslateStorageByUnitAsExpected(string orginalText, string metricUnit, string expectedOutput)
+        public void TextParserExtensionsTranslateStorageByUnitAsExpected(string originalText, string metricUnit, string expectedOutput)
         {
-            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateStorageByUnit(orginalText, metricUnit), expectedOutput));
+            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateStorageByUnit(originalText, metricUnit), expectedOutput));
         }
 
         [Test]
         [TestCase("100k", "100000")]
         [TestCase("100m", "100000000")]
-        public void TextParserExtensionsTranslateNumericUnitAsExpected(string orginalText, string expectedOutput)
+        public void TextParserExtensionsTranslateNumericUnitAsExpected(string originalText, string expectedOutput)
         {
-            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateNumericUnit(orginalText), expectedOutput));
+            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateNumericUnit(originalText), expectedOutput));
+        }
+
+        [Test]
+        [TestCase("1h", "3600")]
+        [TestCase("1hrs", "3600")]
+        [TestCase("1hr", "3600")]
+        [TestCase("1hours", "3600")]
+        [TestCase("1hour", "3600")]
+        //
+        [TestCase("1m", "60")]
+        [TestCase("1minutes", "60")]
+        [TestCase("1minute", "60")]
+        //
+        [TestCase("1000ms", "1")]
+        [TestCase("1000milliseconds", "1")]
+        [TestCase("1000millisecond", "1")]
+        //
+        [TestCase("1000000us", "1")]
+        [TestCase("1000000microsecond", "1")]
+        [TestCase("1000000microseconds", "1")]
+        //
+        [TestCase("1000000000ns", "1")]
+        [TestCase("1000000000nanosecond", "1")]
+        [TestCase("1000000000nanoseconds", "1")]
+        //
+        [TestCase("1s", "1")]
+        [TestCase("1second", "1")]
+        [TestCase("2seconds", "2")]
+        public void TextParserExtensionsTranslateTimeUnitToSecondAsExpected(string originalText, string expectedOutput)
+        {
+            string result = TextParsingExtensions.TranslateToSecondUnit(originalText);
+            Assert.AreEqual(result, expectedOutput);
+        }
+
+        [Test]
+        [TestCase("60s", MetricUnit.Minutes, "1")]
```

Comment:
Great work! I would be more interested in checking more special cases like double(3.14), negative or zeros

Reply:
double and zero are supported. negative time is not.